p2p/service/sim.go

```diff
@@ -172,9 +187,18 @@ func (sn *Node) sendMessageImpl(nodeID p2pcrypto.PublicKey, protocol string, pay
 	return errors.New("could not find " + protocol + " handler for node: " + nodeID.String())
 }
 
+func randSleep(delay uint32) {
+	ranDelay := uint32(0)
+	if delay != 0 {
+		ranDelay = rand.Uint32() % delay
+	}
+	time.Sleep(time.Second*time.Duration(ranDelay))
+}
+
 // Broadcast
 func (sn *Node) Broadcast(protocol string, payload []byte) error {
 	go func() {
+		randSleep(sn.sndDelay)
```

Comment:
does random sleeps in tests is good for us ? if something went wrong it will be hard to know which params caused it.

Reply:
Yes, I talked with Yuval about it. We want to cover as many cases as possible and randomness is a good way to do that. Regarding recreation of the problem - we will have to deal with it later.

Reply:
Please enable non-random sleeps as well. we can't guarantee that all tests that use simulator will need random sleeps. we don't want to make them use randomness if they don't need. (use a boolean of w/e)